csvcubed/csvcubed/schema/cube-config-schema.jsonc

```diff
@@ -0,0 +1,303 @@
+            "type": "string",
+            "format": "uri",
+            "default": "https://tdg.docbook.org/tdg/publishers/5.1/dcterms.html#dcterms.temporal"
+        },
+
+        "columns": {
+            "description": "Columns in the same order as the csv, unspecified column types are assumed to be dimensions, case insensitive",
+			"properties": {
+				"Existing dimension": {"$ref": "#/$defs/existingDimension"},
+				"New dimension": {"$ref": "#/$defs/newDimension"},
+				"Existing attribute": {"$ref": "#/$defs/existingAttribute"},
+				"New attribute": {"$ref": "#/$defs/newAttribute"},
+				"Observation": {"$ref": "#/$defs/observation"},
+				"measures": {"$ref": "#/$defs/measure"},
+				"list of units": {"$ref": "#/$defs/unit"}
+			}
+        }
+    },
+
+    "$defs": {
+		"unitValueTypes": {
+			"type": "object",
+			"oneOf": [
+				{"type": "boolean"},
+				{"Existing unit value from unit column": {"$ref": "#/$defs/unit" }},
+				{"New unit value from unit column": {"$ref": "#/$defs/unit" }}
+			]
+		},
+
+		"unit": {
+			"description": "Definition of units for the observed value",
+			"type": "object",
+			"properties": {
+				"label": {
+					"description": "Replacement label for the Unit",
+					"type": "string"
+				},
+				"from_existing": {
+					"description": "URI of an Existing Qb Dimension",
+					"type": "string",
+					"format": "uri"
+				},
+				"scaling_factor": {
+					"description": "The scaling factor",
+					"type": "decimal"
+
+				},
+				"si_scaling_factor": {
+					"description": "The SI scaling factor",
+					"type": "decimal"
+
+				},
+				"quantity_kind": {
+					"description": "Source URI for the definition of the Unit Kind",
+					"type": "string",
+					"format": "uri"
+				},
+				"definition_uri": {
+					"description": "Source URI for the definition of the Unit",
+					"type": "string",
+					"format": "uri"
+				}
+			}
+		},
+		"measureValueTypes": {
+			"type": "object",
+			"oneOf": [
+				{"type": "boolean"},
+				{"Existing measure value from measure column": {"$ref": "#/$defs/measure" }},
+				{"New measure value from measure column": {"$ref": "#/$defs/measure" }}
+			]
+		},
+
+
+		"dataTypes": {
+			"type": "string",
+			"default": "decimal",
+			"enum": [
+				"anyURI",
+				"boolean",
+				"date",
+				"dateTime",
+				"dateTimeStamp",
+				"decimal",
+				"integer",
+				"long",
+				"int",
+				"short",
+				"nonNegativeInteger",
+				"positiveInteger",
+				"unsignedLong",
+				"unsignedInt",
+				"unsignedShort",
+				"nonPositiveInteger",
+				"negativeInteger",
+				"double",
+				"float",
+				"string",
+				"language",
+				"time"]
+		},
+
+		"existingDimension": {
+            "description": "An existing dimension",
+            "$ref": "#/$defs/dimension"
+        },
+
+        "newDimension": {
+            "description": "A new dimension",
+            "$ref": "#/$defs/dimension",
+            "properties": {
+				"codelist": {
```

Comment:
@canwaf I think we need to correct this clear spelling error ;)

Reply:
Also, both `new` and `existing` dimension should both have the codelist property available on them.

Reply:
Are we joking about "code list" being made "uri-safe" by omitting the space instead of using a dash? Do we need to update our style guide?

Reply:
Okay. I'm fine with this. `code-list` it is.

Reply:
Have updated codelist to code-list, without clear schema differences there is only 'dimension' and not 'new dimension' / 'existing dimension'.